stl/inc/type_traits

```diff
@@ -2513,21 +2512,27 @@ void _Swap_trivial_arrays(_Ty (&_Left)[_Size], _Ty (&_Right)[_Size]) noexcept {
     if constexpr (_Size_parts != 0) {
         const auto _Stop = _Left_ptr + _Size_parts;
         do {
-            unsigned char _Buf[_Part_size_bytes];
-            _CSTD memcpy(_Buf, _Left_ptr, _Part_size_bytes);
-            _CSTD memcpy(_Left_ptr, _Right_ptr, _Part_size_bytes);
-            _CSTD memcpy(_Right_ptr, _Buf, _Part_size_bytes);
+            struct _Buffer_type {
+                unsigned char _Data[_Part_size_bytes];
+            };
+
+            _Buffer_type _Buffer                         = *reinterpret_cast<_Buffer_type*>(_Left_ptr);

```

Comment:
This still violates some restrictions like strict aliasing ([[dcl.ref]/6](https://eel.is/c++draft/dcl.ref#6)), since there're reference bindings in the copy construction/assignment. Although this is probably fine for MSVC. I'm afraid that Clang will possibly "miscompile" this.

Reply:
It doesn't for me, but it indeed can.
I guess I'll have to add `__declspec(may_alias)` (only for Clang, it doesn't exist for MSVC)

Reply:
Added in `__attribute__` form. `[[gnu::]]` form does not work on typedefs, `__declspec` form may be turned on or off.

Reply:
Actually no. clang already emits optimal code for trivial implementation, so we just need to use this on MSVC only.